framework/source/class/qx/io/remote/Rpc.js

```diff
@@ -1072,9 +1102,320 @@ qx.Class.define("qx.io.remote.Rpc",
+            var id = response.id;
+            var method = response.method;
+            var error = response.error;
+
+            // we have the response to our request
+            if (typeof id != "undefined" && typeof method == "undefined") {
+              // we already have a response with the request id
+              if (ids.indexOf(id) !== -1) {
+                ex = new qx.io.remote.exception.Transport(
+                  qx.io.remote.RpcError.type.INVALID_DATA,
+                  "Invalid jsonrpc data: multiple responses with same id.",
+                  {
+                    request : data,
+                    response: response
+                  }
+                );
+                return reject(ex);
+              }
+              ids.push(id);
+              // find original request
+              var originalRequest = this.__requestQueue.find(function (item) {
+                return item.id === id;
+              });
+              // the result id doesn't match any request id
+              if (!originalRequest) {
+                ex = new qx.io.remote.exception.Transport(
+                  qx.io.remote.RpcError.type.INVALID_DATA,
+                  "Invalid jsonrpc data: Unknown request id",
+                  {
+                    request : data,
+                    response: response
+                  }
+                );
+                return reject(ex);
+              }
+              // remove the entry from the queue
+              this.__requestQueue.splice(this.__requestQueue.indexOf(originalRequest), 1);
+              // we have an error
+              if (typeof error != "undefined") {
+                ex = new qx.io.remote.exception.JsonRpc(
+                  error.code,
+                  error.message,
+                  {
+                    request : originalRequest.data,
+                    response: response
+                  }
+                );
+                // inform listeners
+                this.fireDataEvent("error", ex);
+                // reject the individual promise
+                return originalRequest.promise.reject(ex);
+              }
+              // we have our result!
+              else if (typeof result != "undefined") {
+                // inform listeners
+                this.fireDataEvent("success", result);
+                // resolve the individual promise
+                return originalRequest.promise.resolve(result);
+              }
+              // invalid response
+              ex = new qx.io.remote.exception.Transport(
+                qx.io.remote.RpcError.type.INVALID_DATA,
+                "Invalid jsonrpc data for "+ method + ": result is missing.",
+                {
+                  request : originalRequest.data,
+                  response: response
+                }
+              );
+              return originalRequest.promise.reject(ex);
+            }
+            // response is a server call or notification
```

Comment:
What is this? Is this a holdover from the qx1 server thing? I don't think it should be possible nor allowed here, right?

Reply:
I don't understand - can you clarify what you mean?

Reply:
My comment may have been one line too early. I think what I meant was, is it possible for `typeof method` to be undefined in 2.0 (line 1370)?